### ISSUE TITLE
05-cross-sections cache: xscache EAFDataSource, dst_group_struct

### ISSUE THREAD

**Issue by KennellyT** [pyne/pyne#1037]
Hi,
As I was parsing through the 05-cross-sections tutorials after running (Cross Section Caching):
```
xscache = XSCache(group_struct=dst_e_g,
                  data_sources=[EAFDataSource, SimpleDataSource, NullDataSource])
```

I found this bug:
`'EAFDataSource' object has no attribute '_dst_group_struct'`
I have attached the figure of this bug below in figure 1.

I was wondering if this is just a spelling error as the attribute "dst_group_struct" is found in all the files prompted but "**_**dst_group_struct" (note underscore prior to dst) is not?

Thank you.
![cross_section_cache_5_21](https://user-images.githubusercontent.com/31637359/40324263-9f92a552-5cfd-11e8-816f-a4eae2ae0aaf.png)

**Comment by scopatz** [comment]
Thanks for reporting @KennellyT! The leading underscore is likely not a typo, but rather where the DataSources store the `dst_group_struct` internally. This is therefore likely a bug.

**Comment by KennellyT** [comment]
Hey @scopatz,

An update I have is that this bug does not require EAFDataSource to be in XSCache for it to occur. For example see figure 1. I will be looking through pyne.xs.cache.XSCache for a possible fix. 

![image](https://user-images.githubusercontent.com/31637359/40787814-22a4061c-64b4-11e8-8193-3296e0641f97.png)

**Comment by KennellyT** [comment]
Possible find: 
In the other data sources in data_source.py, they have `_dst_group_struct=None` defined within them as seen in class ENDFDataSource(DataSource).
`class ENDFDataSource(DataSource):`
`def __init__(self, fh, src_phi_g=None, dst_group_struct=None, **kwargs):`

 EAFDataSource does not have `_dst_group_struct=None` defined nor mentioned in its class:
`class EAFDataSource(DataSource):`
```
def __init__(self, **kwargs):
        super(EAFDataSource, self).__init__(**kwargs)
```
This could be a potential reason this attribute bug occurs. Figures below show the difference between EAFDataSource and ENDFDataSource 
![eaf_issue](https://user-images.githubusercontent.com/31637359/40789084-3101bed6-64b7-11e8-83e9-ab6522644064.png)
![endf_sol](https://user-images.githubusercontent.com/31637359/40789090-337a23a6-64b7-11e8-8463-2ffb41843077.png)

**Comment by scopatz** [comment]
yeah, that seems to be the issue. But I wonder if the cache is making an incorrect assumption, or if EAF is not implemented correctly.

**Comment by KennellyT** [comment]
Good question @scopatz, I was looking at the structure of data_source.py and saw that dst_group_struct is defined twice as seen in figure 1. I also found this error in issue #810. 
I am going to try and play around with EAF and add an dst_group_struct attribute to see if it fixes the bug.
![image](https://user-images.githubusercontent.com/31637359/40797939-618f639c-64cf-11e8-9455-e5b72999d344.png)
figure 1

**Comment by KennellyT** [comment]
After going through the data sources example from pyne.io and having EAFDataSource with dst_group_struct work and run smoothly, I think this is an issue in cache.
